src/eventsctrl.py

```diff
@@ -1,31 +1,44 @@
-import uuid
 from dateutil import parser
+import json
+import uuid
+
 import web
-from aux import logger_instance, debug_mode
-from api import json, json_response, api_response
-from auth import valid_user, get_uploader_from_token
-from validation import build_params, build_date_clause
-from errors import Error, BADPARAMS
+
 from models.event import Event
 from models.aggregation import Aggregation
 from models.operations import results_to_events
 
+from api import api_response, json_response
+from auth import get_uploader_from_token, valid_user
+from aux import debug_mode, logger_instance
+from cache import redis_client
+from errors import BADPARAMS, Error
+from validation import build_date_clause, build_params
+
+
 logger = logger_instance(__name__)
 web.config.debug = debug_mode()
 
 
-class EventsController():
+class EventsController:
     """Handles work related actions"""
 
     @json_response
     @api_response
-    def GET(self, name):
+    def GET(self):
         """Get Events with various filtering options"""
         event_id = web.input().get('event_id')
 
         if event_id:
-            results = Event.get_from_event_id(event_id)
-            data = results_to_events(results)
+            value = redis_client.get(event_id)
+
+            if value:
+                data = json.loads(value)
```

Comment:
What happens if this data is not able to be parsed as JSON? Are we just to let it fail or what?

Reply:
Yes, technically. We always convert to JSON before saving it to the cache (mostly so that it can be converted to bytes, which is required by redis for caching). I can enforce this functionality in the custom Redis Client though :thinking: